File: src/api/models/exerciseModel.js
const { text } = require("express");
const mongoose = require("mongoose");
const mongoosePaginate = require("mongoose-paginate");
//var autoIncrement = require("mongodb-autoincrement");

const exerciseSchema = new mongoose.Schema({
  name: {
    type: String,
  },
  type: {
    type: String,
  },
  duration: {
    type: String,
  },
  image: {
    type: String,
    default: "exerciseDefault.png",
  },

  description: {
    type: String,
  },
});

exerciseSchema.plugin(mongoosePaginate);

module.exports = mongoose.model("Exercise", exerciseSchema);

File: client/lib/src/exercise/models/exercise_model.dart
class Exercise {
  final dynamic id;
  final String name;
  final String type;
  final String duration;
  final dynamic image;
  final String description;

  Exercise(
      {this.id,
      required this.name,
      required this.type,
      required this.duration,
      required this.image,
      required this.description});

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
        id: json['_id'],
        name: json['name'],
        type: json['type'],
        duration: json['duration'],
        image: json['image'],
        description: json['description']);
  }
}
